internal/pkg/telemetry/disable.go

```diff
@@ -2,10 +2,42 @@ package telemetry
 
 import (
 	"context"
+
+	"go.opentelemetry.io/otel/trace"
 )
 
 const disabledTracingCtxKey = ctxKey("disabled-tracing")
 
+//nolint:gochecknoglobals
+var nopTracer = trace.NewNoopTracerProvider().Tracer("")
+
+// contextTracer wraps trace.Tracer and adds the option to turn off tracing based on the context, see IsTracingDisabled function.
+// In the OpenTelemetry it is not implemented for Go yet: https://github.com/open-telemetry/opentelemetry-specification/issues/530
+type contextTracer struct {
+	wrapped trace.Tracer
+}
```

Comment:
V kode uz mame moznost vypnut telemetriu pomocou `context`, vo vsetkom co je od daneho kontextu odvodene.
Fungovalo to ale iba pre spany generovane priamo v apke, nie pre 3rd party integrations.
Toto riesenie funguje pre vsetky spany, kedze sa wrapuje priamo `TracerProvider`.